test/lifecycle_module_test.dart

```diff
@@ -261,6 +261,11 @@ class TestLifecycleModule extends LifecycleModule {
   }
 }
 
+class ModuleThatNeverUnloads extends LifecycleModule {
+  @override
+  Future<Null> onUnload() => Completer<Null>().future;
```

Comment:
Suggestion (non-blocking): This is probably not a big deal, but I think this Completer will cause this module to be leaked. To avoid it you could put the completer in a public field that we could then call complete on in a teardown.